src/component/utils/exploration/__tests__/wrapBlockNodes.js

```diff
@@ -0,0 +1,580 @@
+  const rootBlock = new ContentBlockNode({
+    key: 'A',
+    text: '',
+    type: 'blockquote',
+    children: List(['B']),
+  });
+
+  const contentState = ContentState.createFromBlockArray([
+    rootBlock,
+    new ContentBlockNode({
+      parent: 'A',
+      key: 'B',
+      text: '',
+      type: 'unordered-list-item',
+      children: List(['C']),
+    }),
+    new ContentBlockNode({
+      parent: 'B',
+      key: 'C',
+      text: 'deeply nested list',
+      type: 'unordered-list-item',
+    }),
+  ]);
+
+  assertWrapBlockNodes({
+    ...PROPS,
+    contentState,
+    block: rootBlock,
+    tree: BlockTree.generate(contentState, rootBlock, null),
+  });
+});
+
+test('renders block with nested child that is of different block type', () => {
+  const rootBlock = new ContentBlockNode({
+    key: 'A',
+    text: '',
+    type: 'blockquote',
+    children: List(['B']),
+  });
+
+  const contentState = ContentState.createFromBlockArray([
+    rootBlock,
+    new ContentBlockNode({
+      parent: 'A',
+      key: 'B',
+      text: 'first list item',
+      type: 'ordered-list-item',
+      children: List(['C']),
+    }),
+    new ContentBlockNode({
+      parent: 'B',
+      key: 'C',
+      text: 'header inside list',
+      type: 'header-one',
+    }),
+  ]);
+
+  assertWrapBlockNodes({
+    ...PROPS,
+    contentState,
+    block: rootBlock,
+    tree: BlockTree.generate(contentState, rootBlock, null),
+  });
+});
+
+test('renders block with nested child that have different wrapperElement', () => {
+  const rootBlock = new ContentBlockNode({
+    key: 'A',
+    text: '',
+    type: 'blockquote',
+    children: List(['B']),
+  });
+
+  const contentState = ContentState.createFromBlockArray([
+    rootBlock,
+    new ContentBlockNode({
+      parent: 'A',
+      key: 'B',
+      text: '',
+      type: 'unordered-list-item',
+      children: List(['C']),
+    }),
+    new ContentBlockNode({
+      parent: 'B',
+      key: 'C',
+      text: 'deeply nested list',
+      type: 'ordered-list-item',
+    }),
+  ]);
+
+  assertWrapBlockNodes({
+    ...PROPS,
+    contentState,
+    block: rootBlock,
+    tree: BlockTree.generate(contentState, rootBlock, null),
+  });
+});
+
+test('renders block with nested children with decorator', () => {
+  const rootBlock = new ContentBlockNode({
+    key: 'A',
+    text: '',
+    type: 'blockquote',
+    children: List(['B']),
+  });
+
+  const contentState = ContentState.createFromBlockArray([
+    rootBlock,
+    new ContentBlockNode({
+      parent: 'A',
+      key: 'B',
+      text: 'first list item',
+      type: 'ordered-list-item',
+      children: List(['C']),
+    }),
+    new ContentBlockNode({
+      parent: 'B',
+      key: 'C',
+      text: 'header inside list',
+      type: 'header-one',
+    }),
+  ]);
+
+  const DecoratedComponent = props => {
+    return <span className={'custom-decorated-text'}>{props.children}</span>;
```

Comment:
nit - I think it can be `className='custom-decorated-text'`, no need for the wrapper curly braces if it's a string.

Reply:
Also it appears that the decorator is never applied in the snapshot output - thought maybe I was missing it, but when I search for 'custom-decorated-text' on this page the only result is here.